conda/models/index_record.py

```diff
@@ -112,3 +113,11 @@ class IndexRecord(DictSafeMixin, Entity):
     link = ComposableField(Link, required=False)
 
     preferred_env = StringField(default=None, required=False, nullable=True)
+
+    @property
+    def combined_depends(self):
+        from .match_spec import MatchSpec
+        result = {ms.name: ms for ms in (MatchSpec(spec) for spec in self.depends or ())}
+        result.update({ms.name: ms for ms in (MatchSpec(spec, option=True)
+                                              for spec in self.constrains or ())})
+        return tuple(itervalues(result))
```

Comment:
@mcg1969 This is the only logic I've added to this PR beyond yours, as we discussed today.  Is using a dict to dedupe by name the best way to do this here?  Do you have other ideas?